.github/workflows/cancel_previous_runs.yml

```diff
@@ -1,18 +0,0 @@
-name: Cancel Duplicate Workflow Runs
```

Comment:
If you're going to remove cancel_previous_runs.yml, you'll also need to remove the wait_cancel_previous job from build_and_test.yml as well as all the dependencies upon it.